### ISSUE TITLE
Jupyter notebook command ignores --notebook-dir=C:\

### ISSUE THREAD

**Issue by pfmoore** [jupyter/notebook#1291]
I was testing out the command to run jupyter notebook to serve notebooks from a specific directory. As a test, I was trying to serve from C:\ (this is clearly not a sensible place to serve from normally, but it was a quick and easy test). But the command ignored the ```--notebook-dir``` option and served from the current directory.

```
    >jupyter notebook --notebook-dir=C:\
    [I 11:05:43.435 NotebookApp] Serving notebooks from local directory: C:\Work\Scratch
    [I 11:05:43.435 NotebookApp] 0 active kernels
    [I 11:05:43.435 NotebookApp] The Jupyter Notebook is running at: http://localhost:8888/
    [I 11:05:43.435 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
    [I 11:05:58.443 NotebookApp] Interrupted...
    [I 11:05:58.443 NotebookApp] Shutting down kernels
```

This is Jupyter 4.0.6, on Windows 7, using Python 3.5.1 64-bit (python.org version).

It appears to only be ```C:\``` that has this behaviour - other directories work fine with or without a trailing backslash, and with back or forward slashes. Also, ```C:/``` works as expected. Oddly, even using ```D:\``` works, so it's somehow specific to the C drive.

It's clearly a very minor corner case, but I wasted quite some time working out what was going on.

**Comment by takluyver** [comment]
Anything informative if you run with `--debug`? Does it do the same if you set it in your config file, or is it specific to the command line option?

**Comment by pfmoore** [comment]
The same behaviour occurs whether the option is set in the config file or on the command line.

Here is the ```--debug``` output.

```
>jupyter notebook --notebook-dir=C:\ --debug
[D 08:40:13.545 NotebookApp] Config changed:
[D 08:40:13.561 NotebookApp] {'MappingKernelManager': {'root_dir': 'C:\\Work\\Scratch'}, 'NotebookApp': {'notebook_dir': 'C:\\', 'log_level': 10}, 'FileContentsManager': {'root_dir': 'C:\\Work\\Scratch'}}
[D 08:40:14.075 NotebookApp] Searching ['C:\\Work\\Scratch', 'C:\\Users\\UK03306\\.jupyter', 'c:\\users\\uk03306\\appdata\\local\\programs\\python\\python35\\etc\\jupyter', 'C:\\ProgramData\\jupyter'] for config files
[D 08:40:14.091 NotebookApp] Looking for jupyter_config in C:\ProgramData\jupyter
[D 08:40:14.091 NotebookApp] Looking for jupyter_config in c:\users\uk03306\appdata\local\programs\python\python35\etc\jupyter
[D 08:40:14.091 NotebookApp] Looking for jupyter_config in C:\Users\UK03306\.jupyter
[D 08:40:14.091 NotebookApp] Looking for jupyter_config in C:\Work\Scratch
[D 08:40:14.091 NotebookApp] Looking for jupyter_notebook_config in C:\ProgramData\jupyter
[D 08:40:14.091 NotebookApp] Looking for jupyter_notebook_config in c:\users\uk03306\appdata\local\programs\python\python35\etc\jupyter
[D 08:40:14.091 NotebookApp] Looking for jupyter_notebook_config in C:\Users\UK03306\.jupyter
[D 08:40:14.091 NotebookApp] Loaded config file: C:\Users\UK03306\.jupyter\jupyter_notebook_config.py
[D 08:40:14.091 NotebookApp] Looking for jupyter_notebook_config in C:\Work\Scratch
[D 08:40:14.091 NotebookApp] Config changed:
[D 08:40:14.091 NotebookApp] {'MappingKernelManager': {'root_dir': 'C:\\Work\\Scratch'}, 'NotebookApp': {'notebook_dir': 'C:\\', 'log_level': 10}, 'FileContentsManager': {'root_dir': 'C:\\Work\\Scratch'}}
[D 08:40:14.262 NotebookApp] Terminals not available (error was No module named 'terminado')
[I 08:40:14.262 NotebookApp] Serving notebooks from local directory: C:\Work\Scratch
[I 08:40:14.262 NotebookApp] 0 active kernels
[I 08:40:14.262 NotebookApp] The Jupyter Notebook is running at: http://localhost:8888/
[I 08:40:14.262 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[D 08:40:14.774 NotebookApp] Using contents: services/contents
[D 08:40:14.899 NotebookApp] 200 GET /tree (127.0.0.1) 124.23ms
[D 08:40:15.287 NotebookApp] 304 GET /custom/custom.css (127.0.0.1) 263.99ms
[D 08:40:15.504 NotebookApp] 200 GET /static/services/contents.js?v=20160406084014 (127.0.0.1) 36.04ms
[D 08:40:15.504 NotebookApp] 200 GET /custom/custom.js?v=20160406084014 (127.0.0.1) 0.00ms
[D 08:40:15.535 NotebookApp] 200 GET /api/config/tree?_=1459928415395 (127.0.0.1) 0.00ms
[D 08:40:15.535 NotebookApp] 200 GET /api/config/common?_=1459928415396 (127.0.0.1) 0.00ms
[D 08:40:15.550 NotebookApp] Native kernel (python3) available from c:\users\uk03306\appdata\local\programs\python\python35\lib\site-packages\ipykernel\resources
[D 08:40:15.550 NotebookApp] Native kernel (python3) available from c:\users\uk03306\appdata\local\programs\python\python35\lib\site-packages\ipykernel\resources
[D 08:40:15.550 NotebookApp] 200 GET /api/kernelspecs (127.0.0.1) 15.53ms
[D 08:40:15.566 NotebookApp] 200 GET /api/sessions?_=1459928415397 (127.0.0.1) 15.53ms
[D 08:40:15.581 NotebookApp] 304 GET /static/components/font-awesome/fonts/fontawesome-webfont.eot? (127.0.0.1) 15.53ms
[D 08:40:15.643 NotebookApp] 200 GET /api/contents?type=directory&_=1459928415398 (127.0.0.1) 62.12ms
```

With ```--noteboog-dir=C:\Work``` I get

```
>jupyter notebook --notebook-dir=C:\Work --debug
[D 08:42:35.746 NotebookApp] Config changed:
[D 08:42:35.762 NotebookApp] {'MappingKernelManager': {'root_dir': 'C:\\Work'}, 'FileContentsManager': {'root_dir': 'C:\\Work'}, 'NotebookApp': {'notebook_dir': 'C:\\Work', 'log_level': 10}}
...
```

Note that in this case, the ```root_dir``` for the ```MappingKernelManager``` and ```FileContentsManager``` is set, whereas when I specify ```C:\``` it isn't.

**Comment by takluyver** [comment]
@minrk worked it out - notebook 4.1 has a regression that doesn't allow setting the notebook directory to root. This was already fixed in #1168, and will be released as notebook 4.2 soon.

**Comment by lvjiujin** [comment]
pfmoore, how great you are,   I change my dir 'c.NotebookApp.notebook_dir = 'c:\notebook'  to c.NotebookApp.notebook_dir = 'c:/notebook' , it works well.
